doc/function_types.md

```diff
@@ -1144,3 +1144,15 @@ including ensmallen:
 This can be useful for situations where you know that the checks should be
 ignored.  However, be aware that the code may fail to compile and give more
 confusing and difficult error messages!
+
+
+## Multi-objective functions
```

Comment:
Can we put this section just after the other function types?  Right now it's after `Alternate matrix types`, which isn't a function type.  I think it would make sense to put it above `Alternate matrix types`, either right after `Semidefinite programs` or possibly before `Constrained functions` (to me it doesn't matter too much which).

Reply:
Okay, on it.